File: java/eyes.selenium.java/src/main/java/com/applitools/eyes/TargetPath.java
package com.applitools.eyes;

import com.applitools.eyes.selenium.ElementReference;
import com.applitools.eyes.selenium.ElementSelector;
import com.applitools.eyes.selenium.RegionLocator;
import com.applitools.eyes.selenium.ShadowDomLocator;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

public class TargetPath {

  public static RegionLocator region(By by) {
    return new RegionLocator(null, new ElementSelector(by));
  }

  public static RegionLocator region(WebElement element) {
    return new RegionLocator(null, new ElementReference(element));
  }

  public static RegionLocator region(String selector) {
    return new RegionLocator(null, new ElementSelector(selector));
  }

  public static ShadowDomLocator shadow(By by) {
    return new ShadowDomLocator(null, new ElementSelector(by));
  }

  public static ShadowDomLocator shadow(WebElement element) {
    return new ShadowDomLocator(null, new ElementReference(element));
  }

  public static ShadowDomLocator shadow(String selector) {
    return new ShadowDomLocator(null, new ElementSelector(selector));
  }

}

File: dotnet/Eyes.Selenium4.DotNet/Fluent/TargetPath.cs
using OpenQA.Selenium;

namespace Applitools.Selenium
{
    public static class TargetPath
    {
        public static RegionLocator Region(By by)
        {
            return new RegionLocator(null, new ElementSelector(by));
        }

        public static RegionLocator Region(IWebElement element)
        {
            return new RegionLocator(null, new ElementReference(element));
        }

        public static RegionLocator Region(string selector)
        {
            return new RegionLocator(null, new ElementSelector(selector));
        }

        public static ShadowDomLocator Shadow(By by)
        {
            return new ShadowDomLocator(null, new ElementSelector(by));
        }

        public static ShadowDomLocator Shadow(IWebElement element)
        {
            return new ShadowDomLocator(null, new ElementReference(element));
        }

        public static ShadowDomLocator Shadow(string selector)
        {
            return new ShadowDomLocator(null, new ElementSelector(selector));
        }
    }
}